systems_cross_building/cross_building.rst

```diff
@@ -193,6 +193,46 @@ The above means that **Conan is able to compile the full graph in a single execu
 the build requires using the ``profile_build`` and then it will compile the libraries using the ``host_profile``
 settings applying the environment of the former ones.
 
+Starting with Conan v1.25 (if the user provides the ``--profile:build``) it is possible to get the relative context
+where a recipe is running during a Conan invocation. The object instatiated from the recipe contains the following
+attributes:
+
+* ``self.settings`` will always contain the settings corresponding to the binary to build/retrieve. It will contain
+  the settings from the profile ``profile_host`` when this recipe appears in the ``host`` context and the settings
+  from the profile ``profile:build`` if this object belongs to the ``build`` context.
+* ``self.settings_build`` will always contain the settings provided in the profile ``profile_build``, even if the
+  recipe appears in the ``build`` context, the build requirements of the build requirements are expected to
+  run in the ``build`` machine too.
+* ``self.settings_target``: for recipes in the ``host`` context this attribute will be equal to ``None``, for those
+  in the ``build`` context, if will depend on the level of anidation: for recipes that are build requirements of 
+  packages in the ``host`` context, this attribute will contain the settins from the profile ``profile_host``, while
+  for recipes that are build requirements of other build requirements the ``self.settings_target`` will contain
+  the values of the ``profile_build``.
+
+With previous attributes, a draft for a recipe that packages a cross compiler could follow this pattern:
+
+.. code-block:: python
+
+    class CrossCompiler(ConanFile):
+        name = "xbuild"
+    
+        settings = "os", "arch", "compiler", "build_type"
+        options = {"target": "ANY"}
+        default_options = {"shared": False, "target": None}
+
+        def configure(self):
+            settings_target = getattr(self, 'settings_target', None)
+            if settings_target is None:
+                # It is running in 'host', so Conan is compiling this package
+                if not self.options.target:
+                    raise ConanInvalidConfiguration("A value for option 'target' has to be provided")
+            else:
+                # It is running in 'build' and it is being used as a BR, 'target' can be inferred from settings
+                if self.options.target:
+                    raise ConanInvalidConfiguration(f"Value for the option 'target' will be computed from settings_target")
+                self.options.target = get_target_from_settings(self.settings_target)
```

Comment:
This ``get_target_from_settings()`` is a bit confusing. If it is an option, I would expect to be in the input. Where is it defined? What it does?

Reply:
It is a function to _get the target [value] from the settings_. I'll write this function before the class with a comment. 

This is the cross compiler example from the repo in https://github.com/jgsogo/conan-xbuild, when this recipe is instantiated in the `build` context then the value for the `target` option CAN be computed from the profile_host (which is the value of `settings_target`).